app/models/media_object.rb

```diff
@@ -546,4 +548,9 @@ def update_permalink_and_dependents
       true
     end
   
+    def remove_bookmarks
+      Bookmark.where(document_id: self.pid).each do |b|
+        b.destroy if ( !User.exists? b.user_id ) or ( Ability.new( User.find b.user_id ).cannot? :read, self )
```

Comment:
I think the second clause here needs to be reworked since a user could have access to an item based upon their session info (virtual_groups, IP, etc.) that isn't being taken into account here.  The result could be you login via LTI and bookmark, the object gets saved and you lose your bookmark, you log back in via LTI and are missing your bookmark.